### PR TITLE
Add agent performance blog post

### PR DESCRIPTION
**Pull request by safchain**

### CODE REVIEW

File: blog/performance-1.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+---
+title: What performance can I expect ?
+layout: blog-post
+author: Sylvain Afchain
+date: 07/02/2019
+---
+
+In this article I'm going to try to answer to a question that often comes about Skydive : "Ok but what will be the impact on my system". Answering this question can be complex as Skydive address multiple use cases. So for this article I'll take a very common use case which is
```
Comment (lebauce) [inline blog/performance-1.md]: comes up about Skydive

File: blog/performance-1.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+monitoring the topology and the interfaces metrics. Here it won't talk about packet capture, We will talk about this aspect in a further article.
+
+## A bit of background
+
+Since the beginning we tried to keep Skydive as a lightweight solution. Almost all the aspect of the monitoring is event based. We use polling only when there is no other solution or when it makes sense to keep performance impacts low. Today only the interface metrics is using polling as
+there is not other solution and makes sense to do sampling.
```
Comment (lebauce) [inline blog/performance-1.md]: and it makes sense to do sampling.

File: blog/performance-1.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+## Let's stress
+
+In order to build the test environment I used a sandbox [script](https://github.com/skydive-project/skydive/blob/master/scripts/scale.sh) that we use to do scaling and profiling test. It can generate "virtual" environment
```
Comment (lebauce) [inline blog/performance-1.md]: profiling tests.

Comment (lebauce) [reply blog/performance-1.md]: can generate a `virtual` environment

File: blog/performance-1.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    <img src="/assets/images/blog/perf-1.png"/>
+  </a>
+</p>
+
+I did the test on my laptop while writing these lines as you will see there is no need to invest to a bigger setup. My laptop is made of 
+1 Intel i7-5600U and 8Go of RAM. 
+
+I used the default metric sample rate which is of 5 seconds. 
+
+Bellow the results of the tests with different density of network namespaces for a given host. As network namespaces are used by containers the results will be really closed from what we would get using Docker containers for example.
```
Comment (lebauce) [inline blog/performance-1.md]: Below

Comment (lebauce) [reply blog/performance-1.md]: closed to what

File: blog/performance-1.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+### First view
+
+Let's do it simple for now, just checking CPU usage per number of network namespace. It as been achieved by getting usage samples each 500ms.
```
Comment (lebauce) [inline blog/performance-1.md]: It has been

File: blog/performance-1.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+As we can see even with 100 network namespace on a host the CPU usage of Skydive is low.
```
Comment (lebauce) [inline blog/performance-1.md]: namespaces

File: blog/performance-1.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+<p class="center">
+  <a href="/assets/images/blog/performance-1-5.png" data-lightbox="Performance 5" data-title="CPU Memory/Object usage">
+    <img src="/assets/images/blog/performance-1-5.png"/>
+  </a>
+</p>
+
+### But with real containers...
+
+Until now I used our "sandbox" script, but what will it be with real container ? For this test I just started 30 containers in
+a loop.
```
Comment (lebauce) [inline blog/performance-1.md]: remove new line

File: blog/performance-1.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+<p class="center" style="display: flex">
+  <a href="/assets/images/blog/performance-1-7.png" data-lightbox="Performance 7" data-title="CPU Usage period">
+    <img src="/assets/images/blog/performance-1-7.png"/>
+  </a>
+  <a href="/assets/images/blog/performance-1-8.png" data-lightbox="Performance 8" data-title="CPU Usage period">
+    <img src="/assets/images/blog/performance-1-8.png"/>
+  </a>
+</p>
+
+The following video shows how the containers are detected and report by Skydive.
```
Comment (lebauce) [inline blog/performance-1.md]: reported

File: blog/performance-1.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+<p>
+  <video poster="" preload="" controls="" loop="" controlslist="nodownload" src="/assets/videos/blog/performance-1-6.webm"></video>
+</p>
+
+## conclusion
+
+The overhead is already low and we can play with the sample rate parameter to low down even more. Of course there are still some work that
```
Comment (lebauce) [inline blog/performance-1.md]: to lower it down even more

Comment (lebauce) [reply blog/performance-1.md]: there are always some work

File: blog/performance-1.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+  </a>
+</p>
+
+The following video shows how the containers are detected and report by Skydive.
+
+<p>
+  <video poster="" preload="" controls="" loop="" controlslist="nodownload" src="/assets/videos/blog/performance-1-6.webm"></video>
+</p>
+
+## conclusion
```
Comment (lebauce) [inline blog/performance-1.md]: Conclusion

File: blog/performance-1.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+
+In this article I'm going to try to answer to a question that often comes about Skydive : "Ok but what will be the impact on my system". Answering this question can be complex as Skydive address multiple use cases. So for this article I'll take a very common use case which is
+monitoring the topology and the interfaces metrics. Here it won't talk about packet capture, We will talk about this aspect in a further article.
+
+## A bit of background
+
+Since the beginning we tried to keep Skydive as a lightweight solution. Almost all the aspect of the monitoring is event based. We use polling only when there is no other solution or when it makes sense to keep performance impacts low. Today only the interface metrics is using polling as
+there is not other solution and makes sense to do sampling.
+
+## Let's stress
```
Comment (lebauce) [inline blog/performance-1.md]: Let's stress it

File: blog/performance-1.md [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+namespaces.
+
+<p>
+  <a href="/assets/images/blog/perf-1.png" data-lightbox="Performance-1" data-title="Skydive performance">
+    <img src="/assets/images/blog/perf-1.png"/>
+  </a>
+</p>
+I did the test on my laptop while writing these lines. As you will see there is no need to invest to a bigger setup. My laptop is made of 1 Intel i7-5600U and 8Go of RAM. 
+
+I used the default metric sample rate which is of 5 seconds.
```
Comment (nplanel) [inline blog/performance-1.md]: I used the default metric sample polling time is 5 seconds.

**Pull request merged**